src/code/examples/movable-camera.rs

```diff
@@ -0,0 +1,373 @@
+#![allow(clippy::too_many_arguments)]
+
+use std::ops::{Div, Mul, Neg};
+
+use bevy::{
+   input::mouse::{MouseMotion, MouseWheel},
+   prelude::*,
+};
+
+// ANCHOR: example
+/// Trait for customizing camera behavior.
+pub trait MovableCameraParams {
```

Comment:
This should just be stored as a resource struct: no reason to overcomplicate things, and that makes it configurable.

Reply:
Or actually, as a bundle of components on the camera entity.

This really wants a real input mapping solution, but that's out-of-scope.

Reply:
oop, you're right, i just didn't know how to do that. i can fix it when I'm awake.